Dev19

svetSanna opened the pull request:
Спринт 19. Навигация BottomNavigationView. Домашняя работа.

Здравствуйте! Буду благодарна за подробные комментарии.

svetSanna committed:
19 спринт
1. Создала RootActivity
2. Перевела SearchActivity в SearchFragment

svetSanna committed:
19 спринт
3. Перевела SettingsActivity в SettingsFragment
4. Перевела MediatekaActivity в MediatekaFragment

svetSanna committed:
19 спринт
6. Сделала selector для bottom_navigation_menu
7. исправила ошибки во фрагментах

svetSanna committed:
19 спринт
8. сделала возвращение по стрелочке из MediaActivity через navigateUp

Nick-review commented on app/src/main/AndroidManifest.xml:
```diff
@@ -16,26 +16,36 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+      <!--  <activity
```
Если код не используется, его лучше удалить, чтобы не мешался, а не комментировать

Nick-review commented on app/src/main/java/com/example/playlistmaker/App.kt:
```diff
@@ -66,26 +66,6 @@ class App : Application() {
         super.onCreate()
     }
 
-    /*fun switchTheme(stringFromSharedPrefs: String?) {
```
Тут аналогично, неиспользуемый код лучше удалять сразу

Nick-review commented on app/src/main/java/com/example/playlistmaker/data/repository/MediaPlayerRepositoryImpl.kt:
```diff
@@ -5,7 +5,6 @@ import com.example.playlistmaker.domain.repository.MediaPlayerRepository
 
 class MediaPlayerRepositoryImpl(private var mediaPlayer : MediaPlayer) :
     MediaPlayerRepository {
-    //private var mediaPlayer = MediaPlayer()
```
Аналогично

Nick-review commented on app/src/main/java/com/example/playlistmaker/data/repository/SearchHistoryRepositoryImpl.kt:
```diff
@@ -5,7 +5,6 @@ import com.example.playlistmaker.domain.entity.Track
 import com.example.playlistmaker.domain.repository.SearchHistoryRepository
 
 class SearchHistoryRepositoryImpl(var searchHistory: SearchHistory) : SearchHistoryRepository {
-    //var searchHistory = SearchHistory()
```
И тут

Nick-review commented on app/src/main/java/com/example/playlistmaker/ui/MediaActivity/MediaActivity.kt:
```diff
@@ -24,6 +26,10 @@ class MediaActivity : AppCompatActivity() {
     //Log.i("MyTest", "MediaActivity.onCreate-4")
```
и тут

Nick-review commented on app/src/main/java/com/example/playlistmaker/ui/MediaActivity/MediaActivity.kt:
```diff
@@ -59,11 +65,19 @@ class MediaActivity : AppCompatActivity() {
         // кнопка "Назад"
```
Как будто в этих комментариях не так много смысла, только усложняют чтения, т.к. из названия button back итак понятно, что это кнопка назад
Есть такое универсальное правило - хороший код в комментариях не нуждается)

Nick-review commented on app/src/main/java/com/example/playlistmaker/ui/MediaActivity/MediaActivity.kt:
```diff
@@ -118,13 +132,6 @@ class MediaActivity : AppCompatActivity() {
                 viewModel.playbackControl()
             }
         }
-        /*else {
```
Не буду писать про комментарии больше, но вы поняли)

svetSanna committed:
19 спринт
9. После ревью. Переделала работу с историей. Все работает, пока я не добавляю в SearchFragment  в функцию onDestroyView() _binding = null

svetSanna committed:
19 спринт
9. После ревью. Переделала работу с историей. Пофиксила

svetSanna committed:
19 спринт
10. После сдачи.
Вернула временно NavigateUp

svetSanna committed:
Вернула все на место
